lib/write_concern.js

```diff
@@ -32,6 +36,36 @@ class WriteConcern {
     }
   }
 
+  static fromUrl(url) {
+    let wc;
+    parseConnectionString(url, (err, parsedUrl) => {
```

Comment:
There is no guarantee that `parseConnectionString` will evaluate and invoke its callback before `wc` is returned. This needs to be rewritten as an async function.